networkx/algorithms/shortest_paths/tests/test_unweighted.py

```diff
@@ -32,6 +34,16 @@ def test_bidirectional_shortest_path(self):
             4, 4, 1, 12, nx.bidirectional_shortest_path(self.grid, 1, 12)
         )
         assert nx.bidirectional_shortest_path(self.directed_cycle, 0, 3) == [0, 1, 2, 3]
+        # test missing source and target
+        target = 8
+        source = 10
+        with pytest.raises(
+            nx.NodeNotFound,
+            match=f"Either source {source} or target {target} is not in G",
+        ):
+            nx.bidirectional_shortest_path(self.cycle, source, target)
```

Comment:
This generally looks good - if we wanted to really bolster the tests here I think we could expand this to check even more cases. I can think of four cases you might want to check:
 1. Only source not in `G`
 2. Only target not in `G`
 3. Neither source nor target in `G` (the case currently covered here)
 4. source == target, but neither in `G`

The last one is relevant because there is another special logic branch to test that `src == target` returns `[src]`. This means that if someone were to reorder the logic in the implementation, it's quite conceivable that they might miss this case and end up breaking that behavior - having a test in place to prevent that would be nice!

The best way to handle all these cases might be to split this out into its own test method and parametrize it, e.g.

```python
    @pytest.mark.parametrize(("src", "tgt"),
        ((8, 3),  # source not in G
         (3, 8),  # target not in G
         (8, 10),  # neither source nor target in G
         (8, 8),  # src == tgt, neither in G
        )
    )
    def test_bidirectional_shortest_path_src_tgt_not_in_graph(src, tgt):
```

Reply:
Thanks alot @rossbar for the review. I will update the pull request soonest